### PR TITLE
QFN-12 footprint for MMZ09332B

### PR DESCRIPTION
**Pull request by drid**
https://www.nxp.com/docs/en/data-sheet/MMZ09332B.pdf

One issue regarding EP. Mechanical has dimensions 1.4mm-1.6mm which result in 1.5mm EP but recommended land pattern has 1.6mm EP
image
image

How is this resolved in the generator?
Also is there a way to create a 3-2-3 thermal via pattern?

### CODE REVIEW

**Comment by chschlue** [comment]
The desired EP size is set using `EP_size_[xy]_overwrite` (see https://github.com/pointhi/kicad-footprint-generator/tree/master/scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON#exposed-pad-handling).
The via pattern is not possible at this time.

**Comment by myfreescalewebpage** [comment]
@drid thanks

 -[ ] lead_width minimum value should be 0.18
- [ ] You should remove all the comment lines

Else @chschlue @poeschlr it seems ok for me but because it's a bit specific I prefer you have a second look

Joel

**Review by chschlue — CHANGES_REQUESTED** [review]
Added line comments.

Anything specific you were asking about, @myfreescalewebpage?

**Comment by myfreescalewebpage** [comment]
@chschlue the footprint name and EP eventually but if my understanding is correct, this is the best we can do to have something closed to the datasheet definition, isn't it ?

**Comment by chschlue** [comment]
Ah. Wasn't paying attention to the name.
Footprint doesn't look very special to me, unless I'm missing something `NXP_` isn't necessary and the size def can be moved to `qfn-1x.yaml`.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/special_qfn.yaml [hunk 1]
```diff
@@ -35,3 +35,61 @@ Texas_X2QFN-12_1.6x1.6mm_P0.4mm:
...
+  #ipc_density: 'least' #overwrite global value for this device.
+  # custom_name_format:
+  body_size_x:
+    minimum: 2.9
+    nominal: 3
+    maximum: 3.1
+  body_size_y:
+    minimum: 2.9
+    nominal: 3
+    maximum: 3.1
```
[the remark this answers is not in the document]
Comment (drid) [reply scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/qfn/special_qfn.yaml:54]: I misinterpreted a 0.1mm dimension. Changing to nominal only

**Review by chschlue — APPROVED** [review]
Thanks.

**Pull request merged**